teacher.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(id, age, specialization, parent_permission: true)
+    super(id, age, name, parent_permission)
```

Comment:
- Same here, since the `id` will be generated automatically by calling `super` you don't need to pass it in the constructor function. Kindly remove it from the constructor and also from the `super()`
- Since you are passing the `name` argument in the `super`, you also need to pass it in the `initialize` constructor, since that is how `super()` will get access to the name. Kindly update your code accordingly.